xcode/Subconscious/Shared/Components/Common/Profile/Sheets/EditProfileSheetModifier.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  EditProfileSheetModifier.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 3/11/2023.
+//
+
+import Foundation
+import ObservableStore
+import SwiftUI
+
+struct EditProfileSheetModifier: ViewModifier {
+    @ObservedObject var app: Store<AppModel>
+    @ObservedObject var store: Store<UserProfileDetailModel>
+    
+    func body(content: Content) -> some View {
+        content
+            .sheet(
+                isPresented: store.binding(
+                    get: \.isEditProfileSheetPresented,
+                    tag: UserProfileDetailAction.presentEditProfile
+                )
+            ) {
+                EditProfileSheet(
+                    store: store.viewStore(
+                        get: \.editProfileSheet,
+                        tag: EditProfileSheetCursor.tag
+                    )
+                )
+            }
+            .onReceive(
+                store.actions.compactMap(AppAction.from),
+                perform: app.send
+            )
+            .onReceive(app.actions, perform: { action in
+                switch (action) {
+                case .completeIndexPeers(let results)
+                    // Only refresh the view if the presented user was indexed
+                    where results.contains(where: { result in
+                        switch (result) {
+                        case .success(let peer) where peer.identity == store.state.user?.did:
+                            return true
+                        default:
+                            return false
+                        }
+                    }):
```

Comment:
This business logic should be handled in a `store` `completeIndexPeers` update function. Do the `contains` logic in there, and no-op if it doesn't match, otherwise return `update(state: state, .refresh(forceSync:false)`.

That way we get a testable update function.

Reply:
Turns out this is stale / deprecated code and we already moved the logic into `UserProfileDetailModel.completeIndexPeers`, deleted.